### PR TITLE
Parse and display block mode and key size

### PR DESCRIPTION
**Pull request by leshi**

### CODE REVIEW

File: u2f-ref-code/java/src/com/google/u2f/server/impl/attestation/android/BlockMode.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  }
+
+  public static BlockMode fromString(String string) throws CertificateParsingException {
+    for (BlockMode mode : BlockMode.values()) {
+      if (mode.toString().equals(string)) {
+        return mode;
+      }
+    }
+
+    throw new CertificateParsingException("Invalid block mode value: " + string);
```
Comment (dongjing-github) [inline u2f-ref-code/java/src/com/google/u2f/server/impl/attestation/android/BlockMode.java]: nit: "Invalid block mode string: "

Comment (leshi) [inline u2f-ref-code/java/src/com/google/u2f/server/impl/attestation/android/BlockMode.java]: Done

**Comment by leshi** [comment]
Thanks, PTAL.

File: u2f-ref-code/java/src/com/google/u2f/server/impl/attestation/android/AuthorizationList.java [hunk 2]
```diff
@@ -77,28 +106,42 @@ public JsonObject toJson() {
...
     public Builder() {
       this.purpose = null;
       this.algorithm = null;
+      this.keySize = null;
+      this.blockMode = null;
     }
 
     public Builder setPurpose(List<Purpose> purpose) {
-      this.purpose = purpose;
+      this.purpose = new ArrayList<Purpose>(purpose);
```
Comment (dongjing-github) [inline u2f-ref-code/java/src/com/google/u2f/server/impl/attestation/android/AuthorizationList.java]: if (purpose != null) {
  this.purpose = new ArrayList<Purpose>(purpose);
}

File: u2f-ref-code/java/src/com/google/u2f/server/impl/attestation/android/AuthorizationList.java [hunk 3]
```diff
@@ -107,8 +150,26 @@ public Builder setAlgorithm(Algorithm algorithm) {
       return this;
     }
 
+    public Builder setKeySize(Integer keySize) {
+      this.keySize = keySize;
+      return this;
+    }
+
+    public Builder setBlockMode(List<BlockMode> blockMode) {
+      this.blockMode = new ArrayList<BlockMode>(blockMode);
```
Comment (dongjing-github) [inline u2f-ref-code/java/src/com/google/u2f/server/impl/attestation/android/AuthorizationList.java]: if (blockMode != null) {
  this.blockMode = new ArrayList<BlockMode>(blockMode);
}

**Comment by dongjing-github** [comment]
lgtm

File: u2f-ref-code/java/src/com/google/u2f/server/impl/attestation/android/AuthorizationList.java (hunk 2 above)
Comment (leshi) [inline u2f-ref-code/java/src/com/google/u2f/server/impl/attestation/android/AuthorizationList.java]: done

File: u2f-ref-code/java/src/com/google/u2f/server/impl/attestation/android/AuthorizationList.java (hunk 3 above)
Comment (leshi) [inline u2f-ref-code/java/src/com/google/u2f/server/impl/attestation/android/AuthorizationList.java]: done

**Pull request merged**